nemo/collections/asr/data/audio_to_text_dali.py

```diff
@@ -0,0 +1,312 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import nvidia.dali as dali
+from nvidia.dali.pipeline import Pipeline
+from nvidia.dali.plugin.pytorch import DALIGenericIterator as DALIPytorchIterator
+from nemo.utils.decorators import experimental
+from omegaconf import DictConfig
+import numpy as np
+import math
+import torch
+
+__all__ = [
+    'AudioToCharDALIDataset',
+]
+
+@experimental
+class AudioToCharDALIDataset(DALIPytorchIterator):
+    """
+    NVIDIA DALI pipeline that loads tensors via one or more manifest files where each line containing a sample descriptor in JSON,
+    including audio files, transcripts, and durations (in seconds).
+    Here's an example:
+    {"audio_filepath": "/path/to/audio.wav", "text_filepath": "/path/to/audio.txt", "duration": 23.147}
+    ...
+    {"audio_filepath": "/path/to/audio.wav", "text": "the transcription", "offset": 301.75, "duration": 0.82, "utt":
+    "utterance_id", "ctm_utt": "en_4156", "side": "A"}
+    Args:
+        manifest_filepath: Path to manifest file with the format described above. Can be comma-separated paths.
+        labels: String containing all the possible characters to map to.
+        sample_rate (int): Sample rate to resample loaded audio to.
+        batch_size (int): Number of samples in a batch.
+        num_threads (int): Number of CPU processing threads to be created by the DALI pipeline.
+        trim_silence (bool): If True, it will extract the nonsilent region of the loaded audio signal.
+        min_duration (float): Determines the minimum allowed duration, in seconds, of the loaded audio files.
+        max_duration (float): Determines the maximum allowed duration, in seconds, of the loaded audio files.
+        shuffle (bool): If set to True, the dataset will shuffled after loading.
+        device (str): Determines the device type to be used for preprocessing. Allowed values are: 'cpu', 'gpu'.
+        device_id (int): Index of the GPU to be used. Only applicable when device == 'gpu'. Defaults to 0.
+        global_rank (int): Worker rank, used for partitioning shards. Defaults to 0.
+        world_size (int): Total number of processes, used for partitioning shards. Defaults to 0.
+        preprocessor_cfg (DictConfig): Preprocessor configuration
```

Comment:
Could you expand the docstring to mention which type of pre-processor it supports (name of class should be sufficient).